### DISCUSSION TITLE
Convert proprietary messages to standard for viewing in 3D panel

### DISCUSSION THREAD

**Discussion by vencivat** [foxglove/community#55] [Q&A] [answered]
Hi all,

Is it possible to create "converter/translator/broker" extension that will subscribe to one topic with propriety message and convert on the fly and publish message to another topic with graphic primitives . 
We want to publish visualization_msgs/MarkerArray messages to draw graphic primitives for custom messages in 3D panel
Something similar is done in http://wiki.ros.org/rviz/Tutorials/Markers%3A%20Basic%20Shapes for RViz

**Comment by jtbandes** [comment, marked as answer]
Yes, this would be a good use case for Node Playground: https://foxglove.dev/docs/studio/panels/node-playground Give it a try and feel free to ask if you run into any roadblocks!

To clarify, Node Playground is something you would configure within your saved layout. As for doing this from an installable extension package, we have an open ticket: https://github.com/foxglove/studio/issues/2212
